scripts/check-one-feature

```diff
@@ -12,21 +12,36 @@ fi
 feature=$1
 shift
 
-if [ ! -d target ]
+# We assume that `target` is the normal target directory.
+base_target_dir="target"
+
+# Check to see if we have the nightly toolchain installed, and if so, do a deeper check to see if
+# there's a custom target directory configured.
+if cargo +nightly 2>&1 >/dev/null; then
+  overridden_base_target_dir=$(cargo +nightly -Zunstable-options config get | grep "build.target-dir" | sed 's/build.target-dir = //' | sed 's/\"//g' || true)
+  if [ ! -z "$overridden_base_target_dir" ]; then
+    base_target_dir=$overridden_base_target_dir
+  fi
+fi
+
+# Make sure that `cargo check` has been run so that we have something to bootstrap our per-feature
+# target directory from.
+if [ ! -d $base_target_dir/debug ]
 then
-  echo $0: 'Run `cargo check` first to prime the `target` directory.'
+  echo "$0: Run \`cargo check\` first to prime the \`${base_target_dir}\` directory."
   exit 1
 fi
 
-target=$PWD/target-feature-$feature
+target=$PWD/$base_target_dir/component-features/feature-$feature
+mkdir -p $target
 log=$target/.logfile
 trap 'rm --force --recursive $target' EXIT
 
-cp --archive --link target $target
-rm --force $target/debug/.cargo-lock
+cp --archive --link $base_target_dir/debug $target
+rm --force $base_target_dir/debug/.cargo-lock
 
 echo "===== Feature: $feature ====="
-if cargo check --workspace --all-targets \
+if cargo check --bin vector \
```

Comment:
Will this still check other targets like `test` without `--all-targets`? Or do we not want that anymore?

Reply:
So, from poking around, I think `cargo check --tests --bin vector` would be the right incantation.

Reply:
As far as "do we not want that anymore?"... no, I believe we _do_ want to check tests.

Given that I believe we're the most common users of feature flags, as the developers, for slimmed builds during dev work to speed up the process... we'd want to ensure feature flags were consistent whether building _or_ testing Vector.

Reply:
Updated in 661743c05.